examples/connect-app/src/main.ts

```diff
@@ -0,0 +1,45 @@
+import { type DialOptions } from '@viamrobotics/rpc/src/dial';
+import * as VIAM from '@viamrobotics/sdk';
+
+async function connect(): Promise<VIAM.ViamClient> {
+  // You can remove this block entirely if your robot is not authenticated.
+  // Otherwise, replace with an actual secret.
+  const secret = '<SECRET>';
+  const credential = {
+    payload: secret,
+    type: 'robot-location-secret',
+  };
+
+  // Replace with the host of your actual robot running Viam.
+  const host = '<HOST>';
+
+  const dialOpts: DialOptions = {
+    authEntity: host,
+    credentials: credential,
+  };
+  return new VIAM.ViamClient(dialOpts);
+}
+
+async function main() {
+  let client: VIAM.ViamClient;
+  try {
+    console.log('app is connecting...');
+    client = await connect();
+    await client.connect();
```

Comment:
For sake of clarity, either the `connect` function defined above should either call the `connect` method on the new ViamClient instance before returning that client instance, or be renamed to `getViamClient` since it does not perform the connection task at the moment.

```ts
client = getViamClient();
await client.connect();
```

or

```ts
async function connect(): Promise<VIAM.ViamClient> {
  // all the other variable setup
  const client = new VIAM.ViamClient(dialOpts);
  await client.connect();
  return client;
}

// within main
client = await connect()
```